pkg/skaffold/sync/sync_test.go

```diff
@@ -973,6 +981,10 @@ func TestPerform(t *testing.T) {
 
 			t.Override(&util.DefaultExecCommand, cmdRecord)
 			t.Override(&client.Client, func(string) (kubernetes.Interface, error) {
+				if test.pod == nil {
+					return fake.NewSimpleClientset(), nil
+				}
+
 				return fake.NewSimpleClientset(test.pod), test.clientErr
 			})
 
```

Comment:
there are no namespaces being given below in Perform the arg is `[]string{""} - why does the first test case still work "no error"? Is there a default namespace being used? 

instead of adding a new test (i.e. TestPerform_WithoutNamespaces) could you just add another test case here? First add a new value to the test struct "namespaces", add to existing tests and then pass that along when calling Perform. And in that test case you'd have an empty namespace.

Reply:
because it's `[]string{""}` slice of 1 item and the item is an empty string, this is why the `len` returns a non-zero value and it works.

Reply:
I was thinking about adding one more test case to the old test, but it'll complicate the test, so decided to create a separate test

Reply:
See above comment it turns out we don't need to check for empty namespaces and can remove the separate test. Appreciate the explanation though.